src/helpers/figures/charts/pie_chart.ts

```diff
@@ -260,6 +261,53 @@ function calculatePercentage(
   return percentage.toFixed(2);
 }
 
+function filterNegativeValues(
+  labels: readonly string[],
+  datasets: readonly DatasetValues[]
+): { labels: string[]; dataSetsValues: DatasetValues[] } {
+  if (datasets.length === 1) {
+    const filteredData: number[] = [];
+    const filteredLabels: string[] = [];
+
+    for (let i = 0; i < labels.length; i++) {
+      const label = labels[i];
+      const dataPoint = datasets[0].data[i];
+
+      if (dataPoint !== undefined && (dataPoint > 0 || isNaN(dataPoint))) {
+        filteredData.push(dataPoint);
+        filteredLabels.push(label || "");
+      }
+    }
+
+    return { labels: filteredLabels, dataSetsValues: [{ ...datasets[0], data: filteredData }] };
+  } else {
+    const numberOfDataPoints = Math.max(
+      labels.length,
+      ...datasets.map((dataset) => dataset.data?.length || 0)
+    );
+
+    const dataPointsIndexes = range(0, numberOfDataPoints).filter((dataPointIndex) => {
+      const label = labels[dataPointIndex];
+      const values = datasets.map((dataset) => dataset.data?.[dataPointIndex]);
+      return label || values.some((value) => value === 0 || (value >= 0 && datasets.length !== 1));
+    });
+
+    const filteredLabels = dataPointsIndexes.map((i) => labels[i] || "");
+    const filteredDatasets = datasets.map((dataset) => ({
+      ...dataset,
+      data: dataPointsIndexes.map((i) => {
+        let dataPoint = dataset.data[i];
+        if (dataPoint !== undefined && dataPoint < 0) {
+          dataPoint = 0; // Replace negative values with 0 when there are multiple datasets
```

Comment:
> don't think it makes sense from a functional POV - if the values of all datasets for the given label are 0 , I think it should be removed from the dataset. Then again, the multi dataset feature is a bit weird to start with ... I will discuss it with François.

Hello @rrahir , It makes sense to remove the data and labels associated with negative data when working with a single dataset because we wouldn't want to display legends for data that is not included in the chart. However, when dealing with multiple datasets, it may not be a good idea to remove the negative data to maintain the integrity of the labels. We can ask @Francois-frgi 's opinion on this matter for a functional POV.

Reply:
take a dataset where ,for a given legend, all corresponding data is non-positive. the label will appear in the legend  but won't be displayed in the graph as the value is set to 0. So there you have it, the label integrity is gone.

Reply:
> take a dataset where ,for a given legend, all corresponding data is non-positive. the label will appear in the legend but won't be displayed in the graph as the value is set to 0. So there you have it, the label integrity is gone.

Oh yes i got it :0,  In that case if all the corresponding data is negative we can remove data ?

Reply:
this is where iI suggested getting @Francois-frgi opinion but personally, I think it's the only strategy that makes sense to be consistent with the "single dataset" case